content/docs/rfcs/51/README.md

```diff
@@ -173,45 +169,80 @@ This example node is part of shards `13`, `14`, and `45` in the Status main-net
-This section merely serves as an outlook.
-A specification of automatic sharding will be added to this document in a future version.
+Autosharding select shards automatically and is the default behaviour for shard choice but opting out is always possible.
+Shards (pubsub topics) not longer have to be chosen, they are instead computed from content topics with the procedure below.
+
+## Rendezvous Hashing
+
+Also known as the Highest Random Weight (HRW) method, has many properties useful for shard selection.
+- Distributed agreement. Local only computation without communication.
+- Load balancing. Content topics are spread as equally as possible to shards.
+- Minimal disruption. Content topics switch shards infrequently.
+- Low overhead. Easy to compute.
+
+### Algorithm
+
+For each shard,
+hash using Sha2-256 the concatenation of
+the content topic `application` field (N UTF-8 bytes),
+`version` (N UTF-8 bytes),
+the cluster index (2 bytes) and
+the shard index (2 bytes)
+take the first 64 bits of the hash,
+divided by 2^64,
+compute the natural logarithm of this number then
+take the negative of the weight (default 1.0) divided by it.
+Finally, sort the values and pick the shard with the highest one.
+
+## Content Topic Prefixes
+So that apps can manipulate the shard selection, 2 prefixes CAN be added to content topics.
+Generation & bias.
+When omitted default values are used.
+
+- Short format `/application/version/subject/encoding`
+- Long format `/generation/bias/application/version/subject/encoding`
+
+### Topic design
+Content topics have 2 purposes filtering and routing.
+Filtering is done by changing the `subject` field as this part is not hashed, it will not affect routing (shard selection).
+The `application` and `version` fields do affect routing.
+Application designer should understand that using different `application` field has a cost.
+It increase the traffic a node has to relay when subscribed to all topics but can benefit nodes that only use a subset of topics.
+
+### Generation
+Monotonously increase and indirectly refer to the total number of shards of the Waku network. 
 
-Automatic sharding is a method for scaling Waku relay in the number of (smaller) content topics.
-It automatically maps Waku content topics to pubsub topics.
-Clients and protocols building on Waku relay only see content topics, while Waku relay internally manages the mapping.
-This provides both scaling as well as removes confusion about content and pubsub topics on the consumer side.
+Default: `0`
 
-From an app point of view, a subscription to a content topic `waku2/xxx` using automatic sharding would look like:
+The first generation (zero) use 8 shards in total.
+This document will be updated with future generation shard numbers.
 
-`subscribe("/waku2/xxx", auto=true)`
+Community consensus should be reach before increasing the generation and shards in the network as it would affect everyone.
```

Comment:
I don't know if this is true? Any application can already use any shard they want. If they agree with our "new generations" they can simply start using that as prefix, but I don't see why consensus is necessary beforehand. Perhaps I'm missing something?

Reply:
Maybe not consensus but I meant that we are going to communicate with the community about changes.

Even that may not be necessary to mention.